nodes/clarify_util.js

```diff
@@ -38,6 +38,15 @@ module.exports = {
     if (payload === undefined) {
       return null;
     }
+    // If the payload is a single number we treat is a real time measurement
+    // and add it to Clarify with the current timestamp.
+    if (_.isFinite(payload)) {
+      let val = payload;
+      payload = {
+        times: [DateTime.now().toString()],
+        values: [val],
+      };
+    }
```

Comment:
This seams magic; it would be easy for end-users to transform the data before sending it to us.

If we do this, we need to properly document all supported input formats; might as well want to support writing a data-frame (no topic) if the goal is to support multiple formats. Might reduce some Node-RED out of memory issues?

Reply:
I think we should support writing data-frame directly yes, as the ability to send more signals in one message has been requested from many users. 
The proposed change here makes it really easy to use a lot of built-in connectors that send the value as a number in msg.payload. 

Agree that it needs to be documented well.

Reply:
I think a lot of the standard blocks also works with things like:
```
time: "timestamp"
value: number
```

That's what I use for the circuits in the TADO example. This is probably _preferred_ because it can then handle the case where you read a time-series and want to preserve the timestamps after writing it through the filter. If we could support this format, but make the `time` optional, that would presumably work well.

Reply:
So, to summarize, we would end up with fields:

- `topic`: Used as Input ID when this information is not available in the payload.
- `signal`: Meta-data for a single signal (require `topic` to be set).
- `signals`: Meta-data for multiple signals, mapped by input ID (ignores `topic`). 
- `payload`: One of:
    - single value `{"time": "timestamp", "value": number}` (require `topic` to be set).
    - multi-value `{"times": ["timestamp"], "value": [number]}` (require `topic` to be set).
    - multi-signal (see DataFrame format in docs) (ignores `topic`)

Reply:
We can do the multi-singal (`signals` field and DataFrame payload support) as a separet task/PR. Just trying to picture how the final docs will end up looking.

Reply:
What about allowing single value without timestamp now, and then add the proposed changes from the comment above in a new PR. These are non-breaking changes.

- Make inputID check less strict
- Allow single value payload
- Fix documentation

Reply:
I know @mogstad has a big rewrite of all the nodes ready that I hope we can get in next week.

Reply:
> What about allowing single value without timestamp now, and then add the proposed changes from the comment above in a new PR. 

According to [this comment](https://github.com/clarify/node-red-contrib-clarify/pull/51/files#r808005969), the implementation change to accept `payload: {value: 42}` rather than `payload: 42`.

Do we want to do this before merging?

Reply:
Also, accept `{time: time, value:42}`

Reply:
FWIW this implementation will never be merged, and the PR is only kept as a reference for now. We can easily implement this in the new implementation, we simply need to agree on the formats to accept.